crates/dids/src/method/mod.rs

```diff
@@ -0,0 +1,31 @@
+pub mod jwk;
+
+use crate::did::Did;
+use crate::resolver::DidResolutionResult;
+use async_trait::async_trait;
+use crypto::key_manager::{KeyManager, KeyManagerError};
+use std::sync::Arc;
+
+/// Errors that can occur when working with DID methods.
+#[derive(thiserror::Error, Debug)]
+pub enum DidMethodError {
+    #[error(transparent)]
+    KeyManagerError(#[from] KeyManagerError),
+    #[error("Failure creating DID: {0}")]
+    DidCreationFailure(String),
+}
+
+/// A trait with common behavior across all DID methods.
+#[async_trait]
+pub trait DidMethod<T: Did, CreateOptions> {
+    const NAME: &'static str;
```

Comment:
What do you think about adding documentation that tells implementers what they should set this constant to, and how this is helpful? I.e. "jwk" or "key" or whatever.

Reply:
I'll add a little blurb here.

I'm going to be adding documentation around how to implement additional DID methods once this, did:key, and did:web land. Was thinking it'll be a README within this `method` module.